zipline/assets/continuous_futures.pyx

```diff
@@ -294,3 +294,22 @@ cdef class OrderedContracts(object):
         for i in range(self._size):
             if sid == sids[i]:
                 return sids[i + offset]
+
+    cpdef long_t[:] active_chain(self, long_t starting_sid, long_t dt_value):
+        cdef Py_ssize_t left, right, i, j
+        cdef long_t[:] sids
+        left = right = 0
+        sids = self.contract_sids
+        start_dates = self.start_dates
+
+        for i in range(self._size):
+            if starting_sid == sids[i]:
+                left = i
+                break
+
+        for j in range(i, self._size):
+            if start_dates[j] > dt_value:
+                right = j
```

Comment:
You are right, if we are backtesting near the end of the historical range. That case would occur. Need to set the `right` value to the last `j` in that case.